### PR TITLE
docs: updated note label syntax in *.md files

### LINKED ISSUE #10458: There are more places in other docs too where this is note is not updated...
There are more places in other docs too where this is not updated...
Maybe we can open an issue for it.

_Originally posted by @parth-gr in https://github.com/rook/rook/pull/10234#discussion_r898109037_

### PR DESCRIPTION
**Pull request by YashasviChaurasia**
Resolves #10458
Updated note label syntax in .md files concerning new guidelines.

Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>

### CODE REVIEW

File: Documentation/Contributing/development-flow.md [hunk 1]
```diff
@@ -287,7 +287,7 @@ Signed-off-by: First Name Last Name <email address>
 
 The `component` **MUST** be in the [list checked by the CI](https://github.com/rook/rook/blob/master/.commitlintrc.json).
 
-Note: sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
+!!! note: sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```
Comment (travisn) [inline Documentation/Contributing/development-flow.md:290]: The `!!! note` must be on its own line to be rendered correctly in the docs, with the content of the note indented. You can search for other places in the docs for consistent formatting.
```suggestion
!!! note
    Sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```

**Review by travisn — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: Documentation/Contributing/development-flow.md (hunk 1 above)
Comment (YashasviChaurasia) [reply Documentation/Contributing/development-flow.md:290]: I have applied and mentioned changed and updated the PR.

File: INSTALL.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ On every commit to PR and master the CI will build, run unit tests, and run inte
 If the build is for master or a release, the build will also be published to
 [dockerhub.com](https://cloud.docker.com/u/rook/repository/list).
 
-> Note that if the pull request title follows Rook's [contribution guidelines](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), the CI will automatically run the appropriate test scenario. For example if a pull request title is "ceph: add a feature", then the tests for the Ceph storage provider will run. Similarly, tests will only run for a single provider with the "cassandra:" and "nfs:" prefixes.
+!!! note that if the pull request title follows Rook's [contribution guidelines](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), the CI will automatically run the appropriate test scenario. For example if a pull request title is "ceph: add a feature", then the tests for the Ceph storage provider will run. Similarly, tests will only run for a single provider with the "cassandra:" and "nfs:" prefixes.
```
Comment (travisn) [inline INSTALL.md:41]: We will need the same formatting change for all of these notes.

File: Documentation/Contributing/development-flow.md [hunk 3]
```diff
@@ -287,9 +287,10 @@ Signed-off-by: First Name Last Name <email address>
 
 The `component` **MUST** be in the [list checked by the CI](https://github.com/rook/rook/blob/master/.commitlintrc.json).
 
-Note: sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
-In that case, it is acceptable to shorten the commit message.
-Also, you don't always need to close an issue, again for a very small fix.
+!!! note
+    sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```
Comment (travisn) [inline Documentation/Contributing/development-flow.md:291]: We need to start the sentence for the note with a capital letter.
```suggestion
    Sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```

**Review by travisn — CHANGES_REQUESTED** [review]
After you make additional changes, please [squash the commits](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-history) so the PR will still have one commit, thanks.

File: Documentation/Contributing/development-flow.md (hunk 3 above)
Comment (YashasviChaurasia) [reply Documentation/Contributing/development-flow.md]: @travisn I am facing issues with codespell check. Could you please guide me through the issue.

Comment (subhamkrai) [reply Documentation/Contributing/development-flow.md]: @YashasviChaurasia I'll create the pr to fix that codespell ci, it is strange that it fails on random pr changes, it is also showing on my pr. For now, I'll suggest if the codespell error in not coming from you changes overlook that

**Review by subhamkrai — APPROVED** [review]

File: INSTALL.md (hunk 2 above)
Comment (YashasviChaurasia) [reply INSTALL.md]: @travisn I have applied the following changes.
Thank you

**Review by parth-gr — CHANGES_REQUESTED** [review]
Please update the commit message
It looks strange, refer : https://rook.io/docs/rook/v1.9/Contributing/development-flow/#commit-structure

File: design/ceph/interacting-with-rook-resources.md [hunk 4]
```diff
@@ -38,6 +38,8 @@ Another example on CephFilesystem with ID=a:
 	app.kubernetes.io/managed-by : "rook-ceph-operator"
 	app.kubernetes.io/created-by : "rook-ceph-operator"
 	rook.io/operator-namespace   : "rook-ceph"
-``` 
+```
+
+!!! note
+	 A totally unique string for an application can be built up from (a) app.kubernetes.io/component, (b) app.kubernetes.io/part-of, (c) the resource's namespace, (d) app.kubernetes.io/name, and (e) app.kubernetes.io/instance fields. For the example above, we could join those fields with underscore connectors like this: cephclusters.ceph.rook.io_rook-ceph_rook-ceph_ceph-mon_a. Note that this full spec can easily exceed the 64-character limit imposed on Kubernetes labels.
```
Comment (galexrt) [inline design/ceph/interacting-with-rook-resources.md:44]: While you are at it, could you add backticks around these values to make it easier to read it.
```suggestion
	 A totally unique string for an application can be built up from (a) `app.kubernetes.io/component`, (b) `app.kubernetes.io/part-of`, (c) the resource's namespace, (d) `app.kubernetes.io/name`, and (e) `app.kubernetes.io/instance` fields. For the example above, we could join those fields with underscore connectors like this: `cephclusters.ceph.rook.io_rook-ceph_rook-ceph_ceph-mon_a`. Note that this full spec can easily exceed the 64-character limit imposed on Kubernetes labels.
```

**Review by galexrt — APPROVED** [review]
Besides your commit message, the doc changes LGTM

File: INSTALL.md [hunk 5]
```diff
@@ -38,7 +38,8 @@ On every commit to PR and master the CI will build, run unit tests, and run inte
 If the build is for master or a release, the build will also be published to
 [dockerhub.com](https://cloud.docker.com/u/rook/repository/list).
 
-> Note that if the pull request title follows Rook's [contribution guidelines](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), the CI will automatically run the appropriate test scenario. For example if a pull request title is "ceph: add a feature", then the tests for the Ceph storage provider will run. Similarly, tests will only run for a single provider with the "cassandra:" and "nfs:" prefixes.
+!!! note
+    That if the pull request title follows Rook's [contribution guidelines](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), the CI will automatically run the appropriate test scenario. For example if a pull request title is "ceph: add a feature", then the tests for the Ceph storage provider will run. Similarly, tests will only run for a single provider with the "cassandra:" and "nfs:" prefixes.
```
Comment (travisn) [inline INSTALL.md:42]:
```suggestion
    If the pull request title follows Rook's [contribution guidelines](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure), the CI will automatically run the appropriate test scenario. For example if a pull request title is "ceph: add a feature", then the tests for the Ceph storage provider will run. Similarly, tests will only run for a single provider with the "cassandra:" and "nfs:" prefixes.
```

**Review by travisn — CHANGES_REQUESTED** [review]
Please simplify the commit message so it only mentions the changes, for example this would be a good commit message:
```
docs: updated note label syntax in *.md files

The note label is updated in several places in the docs for the improved rendering.

Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>
```
```

**Comment by YashasviChaurasia** [comment]
I think I should create a new PR as I have created quite a lot of mess around here just to accommodate small changes.
I will close this PR before uploading a new one.

**Comment by travisn** [comment]
> I think I should create a new PR as I have created quite a lot of mess around here just to accommodate small changes. I will close this PR before uploading a new one.

Wait, before closing this and opening a new PR, try squashing the commits to a single commit. There are many articles online for squashing.

**Comment by parth-gr** [comment]
> I think I should create a new PR as I have created quite a lot of mess around here just to accommodate small changes. I will close this PR before uploading a new one.

You can follow these easy steps https://stackoverflow.com/questions/5189560/how-do-i-squash-my-last-n-commits-together

**Comment by YashasviChaurasia** [comment]
The problem I am facing is that I have already made changes to a few files in the branch I submitted for PR.
But now, just squashing the last n branches is not changing those files as they are not being updated in the latest commit.
hence the commit messages for those previous commits won't change.

**Comment by YashasviChaurasia** [comment]
I have tried updating the commit messages.

**Comment by parth-gr** [comment]
> The problem I am facing is that I have already made changes to a few files in the branch I submitted for PR. But now, just squashing the last n branches is not changing those files as they are not being updated in the latest commit. hence the commit messages for those previous commits won't change.

I can't understand what you want to say,

Do run these steps and Let me know if they work,
1)  Interactive reset
> `git rebase -i HEAD~7`
> An interactive window will be opened replace `pick` with `s` for all the commits just leave the top commit of yours
> close and save the changes

2) Reset Hack

> git reset --soft HEAD@{6}
> git add . && git commit -s --amend

**Comment by parth-gr** [comment]
> I have tried updating the commit messages. for some reason, DCO is failing even after signing my commits.

Don't worry looks like your first PR in rook, try to solve it, you will learn a lot
Take your time!

**Comment by parth-gr** [comment]
> > I have tried updating the commit messages. for some reason, DCO is failing even after signing my commits.
> 
> Don't worry looks like your first PR in rook, try to solve it, you will learn a lot Take your time!

For DOC I guess it's probably your signed not right,
As I see the space in between your username I never that before, try to fix that maybe that's the reason

**Comment by travisn** [comment]
Now there are 22 commits. Try `git rebase upstream master` and then `git push origin issue#10458_ -f` to see if that helps clear it out.

**Comment by YashasviChaurasia** [comment]
I have created a new PR as the current PR was broken beyond resolve squashing commits made the PR more of a mess and more and more checks started to fail.
I am closing this current PR and have updated it with a new one.

**Pull request closed without merging**